mendota_ice/src/plot_utils.R

```diff
@@ -0,0 +1,144 @@
+    this_year <- filter(ice_data, start == year, !is.na(n_days))
+    year_col <- this_year$col
+    if (!is.na(year_col)){
+      year_col <- substr(year_col, start = 1, stop = 7)
+      rect(xleft = year+0.5, ybottom = 18, xright = year+1.5, ytop = 170, col = paste0(year_col, col_suf[1]), border = NA)
+      
+    }
+    col_suf <- tail(col_suf, -1L)
+  }
+  
+  col_suf <- c("FF","BF","99","80", "73")
+  cexs <- c(7,6,5.5,5,4.5)
+  for (year in years){
+    this_year <- filter(ice_data, start == year, !is.na(n_days))
+    year_col <- this_year$col
+    if (!is.na(year_col)){
+      year_col <- substr(year_col, start = 1, stop = 7)
+      points(year+1, this_year$n_days, pch = 21, col = paste0(year_col, col_suf[1]), 
+             bg = paste0(year_col, col_suf[1]), cex = cexs[1])  
+    }
+    col_suf <- tail(col_suf, -1L)
+    cexs <- tail(cexs, -1L)
+  }
+  
+  
+  
+  
+  ice_blocks <- filter(ice_data, abs(as.numeric(close_date-belongs_to_yr)) < 100, abs(as.numeric(open_date-belongs_to_yr)) < 150)
+  
+  # new_mai <- current_mai <- par()$mai
+  # new_mai[3] <- 0.5
+  # par(mai = new_mai)
+  xlim <- c(belongs_to_yr-50, belongs_to_yr+140)
+  par(mai = c(0.6, 0.1, 0.85, 0))
+  plot(0, NA, xlim = xlim, 
+       ylim = c(0,1.48), xaxs = 'i', yaxs = 'i', ylab = "", xlab = "", axes = FALSE, xpd = FALSE)
+  
+  # start ticks Nov 1
+  tick_spots <- seq(belongs_to_yr-61, by = 'month', length.out = 12)
+  tick_spots[5] <- tick_spots[4]+29
+  tick_spots[6] <- tick_spots[5]+31
+  tick_spots[7] <- tick_spots[6]+30
+  tick_lab <- format(tick_spots, '%b')
+  tick_lab[3] <- paste0(tick_lab[3],'-    ')
+  axis(1, at = tick_spots, labels = tick_lab, tick = TRUE, lwd = 2.5)
+  axis(1, at = belongs_to_yr+6.5, labels = sprintf("%s", format(belongs_to_yr, '%Y')), 
+       tck = 0, lwd = NA, cex.axis = 3.5, font = 2)
+  
+  
+  for (j in 1:nrow(ice_blocks)){
+    this_block <- ice_blocks[j, ]
+    rect(xleft = this_block$close_date, ybottom = 0.2, xright = this_block$open_date, 1.2, 
+         col = '#82B4B7') #4392F1
+    rect(xleft = this_block$close_date, ybottom = 0.2, xright = this_block$open_date, 1.2, 
+         col = 'white', density = 10, lwd = 3)#4392F1
+  }
+  par(xpd = TRUE)
+  text(belongs_to_yr-48, 1.55, 'Ice duration', cex = 4, pos = 4)
+  mtext(text = 'Data from Wisconsin State Climatology Office; blue bars are 10 years with longest ice-cover, red bars are the shortest 10', 
+        side = 1, line = 3, at = belongs_to_yr+45, cex = 1.9, padj = 2.2, col = 'grey40')
+  showtext_end()
+  dev.off()
+}
+
+create_frame_plan <- function(dates){
+  step1 <- create_task_step(
+    step_name = 'plot',
+    target_name = function(task_name, step_name, ...) {
+      sprintf('figures/mendota_ice_%s101.png', task_name)
+    },
+    command = "plot_ice_frame(target_name, ice_data)"
+  )
+  task_plan <- create_task_plan(as.character(dates), list(step1),
+                                final_steps='plot', add_complete = FALSE)
+  return(task_plan)
+}
```

Comment:
I guess my only concern with this pattern is that now the resulting task plan does not include the most important task of all - the final combining one. This breaks the pattern of the task plan describing all steps and being ready to be turned into a makefile as-is. Is there any way to add this combiner pattern to `create_task_plan` instead?

Reply:
Hmmm...Is that more of a question/comment for `scipiper`? I didn't know the purpose of the task_plan was to be standalone (and include combiners).

Reply:
Oh, maybe. I guess I thought this was an example of that feature and that I should be commenting on the general combiner pattern/implementation. I see now that it is in scipiper but still a branch and not in the main version.

As far as using what is implemented in that branch of scipiper, this is straightforward to me!